pallets/msa/src/lib.rs

```diff
@@ -481,17 +545,14 @@ pub mod pallet {
 					)?;
 					Ok(())
 				})?;
-
-			Self::deposit_event(Event::MsaCreated {
-				msa_id: new_delegator_msa_id,
-				key: new_delegator_public_key,
-			});
-
+			let event =
+				Event::MsaCreated { msa_id: new_delegator_msa_id, key: new_delegator_public_key };
+			Self::index_event(event.clone());
```

Comment:
You may want to index events after they are actually deposited.  It shouldn't make a difference but if there was some sort of failure, it could lead to incorrect data.

Reply:
i think its irrelevant, we should not be expecting this part to fail in any situation